### ISSUE TITLE
TypeError in protocol.py @ line "self.device_worn_confidence = 1 - (status_info & 3)/3"

### ISSUE THREAD

**Issue by prakash1895** [labstreaminglayer/App-Zephyr#7]
Sometimes the status_info returns a NaN value, which is of float datatype. So, the '&' operation at line "self.device_worn_confidence = 1 - (status_info & 3)/3" in protocol.py throws a TypeError:

Error Msg:
TypeError: unsupported operand type(s) for &: 'float' and 'int'

**Comment by chkothe** [comment]
Good catch! It looks like one could check for `status_info` being NaN early in that `_decode_status_info` method and then set all the fields that depend on it to NaN instead of going through the integer operations. Something like 
```
isnan = status_info != status_info  # checks if NaN
self.device_worn_confidence = math.nan if isnan else 1 - (status_info & 3)/3
... # same for other fields
```

I don't have the device at hand to test that at this time, but are you able to test this code (and if possible submit a PR)?

**Comment by prakash1895** [comment]
Thanks, the fix worked, but I encountered another TypeError @ line 462 in protocol.py

Error:
self.br_signal_eval_state = (ext_status_info >> 5) & 3 + flags_valid
TypeError: unsupported operand type(s) for &: 'int' and 'float'

The fix, which worked for me:

try:
    self.br_signal_eval_state = (ext_status_info >> 5) & 3 + flags_valid
except TypeError:
    self.br_signal_eval_state = int((ext_status_info >> 5) & 3) + flags_valid

**Comment by chkothe** [comment]
Ok, thanks for confirming!

That the second fix works is actually a bit surprising (should give another type error, also feels maybe a bit verbose, even if it works).

However, the good thing is that for this  particular flags collection (`ext_status_info`), we already have `flags_valid`, which is either 0 or nan, and which is added to each of the fields to ensure that they correctly take on a nan value if they're invalid. The only problem is that, in addition to that, we write `ext_status_info = parse_num(payload[69:71], False, inval=0xFFFF)` which will make the entire `ext_status_info` a nan if it matches `0xFFFF`, causing the & and >> to fail. 

So I think what should also work would be to just change that line to `ext_status_info = parse_num(payload[69:71])` and leave all the rest unchanged (i.e., without try/catch). However, it'd be good to get that tested too.
